src/hypothesis/internal/compat.py

```diff
@@ -290,16 +290,8 @@ def inner(*args, **kwargs):
         getfullargspec = silence_warnings(getfullargspec)
 
 
-def get_type_hints(thing):
-    """Try to return any type hints for ``thing``."""
-    try:
-        import typing
-        return typing.get_type_hints(thing)
```

Comment:
That's the `except AttributeError` case, IIRC.

Alright.  Let's merge as-is, and if anyone actually complains about inference we can consider more of a side-port then.